plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt

```diff
@@ -64,19 +63,17 @@ internal fun Configuration.isRelevant(): Boolean {
 /**
  * Return a map that associates names of artifact repositories to their model representations.
  */
-internal fun RepositoryHandler.associateNamesWithUrlsTo(repositories: MutableMap<String, OrtRepository?>) =
-    associateTo(repositories) { it.name to it.toOrtRepository() }
+internal fun RepositoryHandler.associateNamesWithUrlsTo(repositories: MutableMap<String, UrlArtifactRepository?>) =
+    associateTo(repositories) { it.name to it as? UrlArtifactRepository }
```

Comment:
While at it, I believe we should generally avoid maps with nullable values, as a lookup then cannot distinguish between the key not being present, or the value being null. So, would this work?

```kotlin
internal fun RepositoryHandler.associateNamesWithUrlsTo(repositories: MutableMap<String, UrlArtifactRepository>) =
    filterIsInstance<UrlArtifactRepository>().associateByTo(repositories) { (it as ArtifactRepository).name }
```

Reply:
Yes, I also found this construct with the map with nullable values suspicious. Changed as proposed.